source/__tests__/integration/create-ontologies.test.js

```diff
@@ -3,43 +3,55 @@
 describe('Create profile has ontologies available for resource template', () => {
   const modalSel = 'div#chooseResource > div.modal-dialog > div.modal-content'
   beforeAll(async () => {
-    // page.on('console', msg => console.log('PAGE LOG:', msg.text()))
     await page.goto('http://127.0.0.1:8000/#/profile/create/')
-    page
-      .waitForSelector('a#addResource')
-      .then(async () => await page.click('a#addResource'))
-      .catch(error => console.log(`promise error for addResource link: ${error}`))
-    page
-      .waitForSelector('a#resourceChoose')
-      .then(async () => await page.click('a#resourceChoose'))
-      .catch(error => console.log(`promise error for select resource link: ${error}`))
-    await page.waitForSelector(modalSel)
+    await page.waitForSelector('a#addResource')
+    await page.click('a#addResource')
+    await page.waitForSelector('a#resourceChoose')
+    await page.click('a#resourceChoose')
+    return await page.waitForSelector(modalSel)
```

Comment:
Can you say more about the pattern of having a `return` on the last line of a before/after? Maybe it's related to the pattern I wrote up? Or not? Would having a bare `return` on the line following that last `await` statement accomplish the same thing? I guess the tests fail (or false positive) without this `return`?

Reply:
This is based on your pattern, yes, and on https://jestjs.io/docs/en/setup-teardown.  Returning a promise is a legit way setup and teardown blocks to block until the async is done.  I guess I assumed it would always do the previous lines before the last one ... is that wrong?   All of the jest examples are for single line things.